src/v/model/fundamental.h

```diff
@@ -286,7 +286,7 @@ struct ntp {
     topic_partition tp;
 
     bool operator==(const ntp& other) const {
-        return ns == other.ns && tp == other.tp;
+        return tp == other.tp && ns == other.ns;
```

Comment:
Same reasoning as above: the namespace is almost _always_ a match, so compare it last.